src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2ApplicationTest.kt

```diff
@@ -289,6 +291,127 @@ class Cas2ApplicationTest : IntegrationTestBase() {
+              }.id,
+            )
+          }
+
+          // create 4 x un-submitted in-progress applications for this user
+          repeat(4) {
+            unSubmittedIds.add(
+              cas2ApplicationEntityFactory.produceAndPersist {
+                withApplicationSchema(applicationSchema)
+                withCreatedByUser(userEntity)
+                withCrn(offenderDetails.otherIds.crn)
+                withData("{}")
+              }.id,
+            )
+          }
+
+          username = userEntity.nomisUsername
+        }
+      }
+    }
+
+    @Test
+    fun `returns all applications for user when isSubmitted is null`() {
+      val jwt = jwtAuthHelper.createValidNomisAuthorisationCodeJwt(username!!)
+
+      val rawResponseBody = webTestClient.get()
+        .uri("/cas2/applications")
+        .header("Authorization", "Bearer $jwt")
+        .header("X-Service-Name", ServiceName.cas2.value)
+        .exchange()
+        .expectStatus()
+        .isOk
+        .returnResult<String>()
+        .responseBody
+        .blockFirst()
+
+      val responseBody =
+        objectMapper.readValue(rawResponseBody, object : TypeReference<List<Cas2ApplicationSummary>>() {})
+
+      val uuids = responseBody.map { it.id }.toSet()
+      Assertions.assertThat(uuids).isEqualTo(submittedIds.union(unSubmittedIds))
+    }
+
+    @Test
+    fun `returns submitted applications for user when isSubmitted is true`() {
+      val jwt = jwtAuthHelper.createValidNomisAuthorisationCodeJwt(username!!)
+
+      val rawResponseBody = webTestClient.get()
+        .uri("/cas2/applications?isSubmitted=true")
+        .header("Authorization", "Bearer $jwt")
+        .header("X-Service-Name", ServiceName.cas2.value)
+        .exchange()
+        .expectStatus()
+        .isOk
+        .returnResult<String>()
+        .responseBody
+        .blockFirst()
+
+      val responseBody =
+        objectMapper.readValue(rawResponseBody, object : TypeReference<List<Cas2ApplicationSummary>>() {})
+
+      val uuids = responseBody.map { it.id }.toSet()
+      Assertions.assertThat(uuids).isEqualTo(submittedIds)
+    }
+
+    @Test
+    fun `returns unsubmitted applications for user when isSubmitted is false`() {
+      val jwt = jwtAuthHelper.createValidNomisAuthorisationCodeJwt(username!!)
+
+      val rawResponseBody = webTestClient.get()
+        .uri("/cas2/applications?isSubmitted=false")
+        .header("Authorization", "Bearer $jwt")
+        .header("X-Service-Name", ServiceName.cas2.value)
+        .exchange()
+        .expectStatus()
+        .isOk
+        .returnResult<String>()
+        .responseBody
+        .blockFirst()
+
+      val responseBody =
+        objectMapper.readValue(rawResponseBody, object : TypeReference<List<Cas2ApplicationSummary>>() {})
+
+      val uuids = responseBody.map { it.id }.toSet()
+      Assertions.assertThat(uuids).isEqualTo(unSubmittedIds)
+    }
+
+    private fun randomDatetimeWithin30Days(): OffsetDateTime {
```

Comment:
could you use `OffsetDateTime.now().randomDateTimeBefore(30)` instead of this custom function?

Reply:
Thanks didn't know this existed in `RandomData.kt`.  Have used it.